packages/docusaurus-plugin-content-docs/src/versions.ts

```diff
@@ -183,7 +182,9 @@ function getVersionMetadataPaths({
     versionName,
   });
 
-  const sidebarFilePath = isCurrentVersion
+  const sidebarFilePath = !options.sidebarPath
```

Comment:
I don't think this condition is correct, the "main test"  should remain based on the `isCurrentVersion`. Versioned docs should always have a sidebar file available

Reply:
Hmm actually not: versioned files can have no sidebars file, so we may compute a path for them but in the end the file may not exist and we shouldn't warn.

This will be better:

```js
  function getSidebarFilePath() {
    if (isCurrentVersion) {
      return options.sidebarPath ? path.resolve(context.siteDir, options.sidebarPath) : options.sidebarPath
    } else {
      return path.join(
        getVersionedSidebarsDirPath(context.siteDir, options.id),
        `version-${versionName}-sidebars.json`,
      );
    }
  }
```